src/context/display/deprecationCheck.ml

```diff
@@ -22,11 +22,9 @@ let warned_positions = Hashtbl.create 0
 let warn_deprecation dctx s p_usage =
 	let pkey p = (p.pfile,p.pmin) in
 	if not (Hashtbl.mem warned_positions (pkey p_usage)) then begin
-		Hashtbl.add warned_positions (pkey p_usage) (s,p_usage);
-		if not (is_diagnostics dctx.com) then begin
```

Comment:
Hmm, there's a chance that this was here for a reason, though I can't immediately tell why.

Reply:
Yeah, I'll try to figure out why it was there. I had to remove it, though, otherwise the module could be cached without the warning if cached from diagnostics.. :/

Reply:
This originally came from https://github.com/HaxeFoundation/haxe/pull/8632